bin/pasta_form_patchwork_relations.py

```diff
@@ -0,0 +1,62 @@
+import sys
+import os
+
+from logging import getLogger
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+from pypasta import *
+
+log = getLogger(__name__[-15:])
+
+
+def form_patchwork_relations(config, prog, argv):
+    parser = argparse.ArgumentParser(prog=prog,
+                                     description='Form patch groups using Patchwork IDs instead of Message IDs')
+    parser.add_argument('-infile', metavar='infile', default=None,
+                        help='The patch groups file to be used as input'
+                             'If not specified will default to value of PATCH_GROUPS config variable')
+
+    args = parser.parse_args(argv)
+
+    if config.mode != config.Mode.MBOX:
+        log.error('Only works in Mbox mode!')
+        return -1
+
+    config.repo.register_mbox(config)
+    mbox = config.repo.mbox
+
+    if args.infile is None:
+        f_clustering = config.f_clustering
+    else:
+        f_clustering = args.infile
+
+    f_result = 'patch-groups-patchwork'
+
+    if not os.path.exists(f_clustering):
+        log.error('Patch groups file does not exist!')
+        return -1
+
+    with open(f_clustering, 'r') as f:
+        f = f.read().split('\n')
+
+    while len(f) and not f[-1]:
+        f.pop()
+
+    f = [x.split(' ') for x in f]
+
+    for line in f:
+        new_line = list()
+        count = 0
+        for item in line:
+            if item == '=>':
+                break
+            patch = mbox.get_patchwork_ids(item)
+            new_line += patch
+            count += 1
+        line[:count] = new_line
+
+    with open(f_result, 'w+') as output:
+        result = list()
+        for line in f:
+            result.append(' '.join([str(x) for x in line]))
+        output.write('\n'.join(result) + '\n')
```

Comment:
No, please don't do it this way. We have an API for this kind of stuff. You're reintroducing the patch-groups parser which we already have inside pypasta/Clustering.py.

What you should rather do is something like:

```
clustering = config.load_cluster(...)
clustering_patchwork = Clustering()

for downstream, upstream in clustering.iter_split():
    patchwork_ids = {mbox.get_patchwork_ids(p) for p in downstream)
    clustering_patchwork.insert(patchwork_ids)
    ...

clustering_patchwork.to_file(args.infile)
```

Reply:
Yes, using this API makes more sense!